src/rust/grapl-web-ui/src/routes/api/plugin/get_deployment.rs

```diff
@@ -0,0 +1,96 @@
+use actix_web::{
+    web,
+    HttpResponse,
+    Responder,
+};
+use rust_proto::graplinc::grapl::api::plugin_registry::v1beta1::{
+    GetPluginDeploymentRequest,
+    PluginDeploymentStatus,
+    PluginRegistryServiceClient,
+};
+
+use super::PluginError;
+
+#[derive(serde::Deserialize)]
+pub(super) struct GetDeploymentParameters {
+    plugin_id: uuid::Uuid,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct PluginDeploymentResponse {
+    pub plugin_id: uuid::Uuid,
+    pub timestamp: std::time::SystemTime,
+    #[serde(serialize_with = "serialize_deployment_status")]
+    #[serde(deserialize_with = "deserialize_deployment_status")]
+    pub status: PluginDeploymentStatus,
+    pub deployed: bool,
+}
+
+#[tracing::instrument(skip(plugin_registry_client, data))]
+pub(super) async fn get_deployment(
+    plugin_registry_client: web::Data<PluginRegistryServiceClient>,
+    user: crate::authn::AuthenticatedUser,
+    data: web::Query<GetDeploymentParameters>,
+) -> Result<impl Responder, PluginError> {
+    let requested_plugin_id = data.plugin_id;
+
+    let mut plugin_registry_client = plugin_registry_client.get_ref().clone();
+
+    super::verify_plugin_ownership(&mut plugin_registry_client, &user, requested_plugin_id).await?;
+
+    let request = GetPluginDeploymentRequest::new(requested_plugin_id);
+    let response = plugin_registry_client
+        .get_plugin_deployment(request)
+        .await?
+        .plugin_deployment();
+
+    let web_response = PluginDeploymentResponse {
+        plugin_id: response.plugin_id(),
+        timestamp: response.timestamp(),
+        status: response.status(),
+        deployed: response.deployed(),
+    };
+
+    Ok(HttpResponse::Ok().json(web_response))
+}
+
+//TODO: It'd be great if rust_proto::graplinc::grapl::api::plugin_registry::v1beta1::PluginDeploymentStatus
```

Comment:
This idea scares me a little bit because all the stuff in `rust_proto::graplinc::grapl::api` is "internal" and not deliberately exposed publicly. In this particular case it's just some innocuous enum, but that could change later and the person changing it might not know this thing is actually public. 

I think making the public API code very deliberate about what it does expose, and ensuring by default it _does not expose_ internal data is much safer.